pkg/cmd/roachtest/tests/db_console_endpoints.go

```diff
@@ -0,0 +1,419 @@
+				if err := testEndpoint(ctx, client, baseURL, ep, idMap[nodeID].String(), ep.getVerifyResponse(), l, multiVersionTest); err != nil {
+					return errors.Wrapf(err, "failed testing endpoint %s with own node ID", ep.url)
+				}
+				// Test with another node ID.
+				otherNodeID := (nodeID % len(idMap)) + 1
+				if err := testEndpoint(ctx, client, baseURL, ep, idMap[otherNodeID].String(), ep.getVerifyResponse(), l, multiVersionTest); err != nil {
+					return errors.Wrapf(err, "failed testing endpoint %s with other node ID", ep.url)
+				}
+			}
+		} else {
+			// For endpoints without node IDs, test on each node.
+			for nodeID := 1; nodeID <= len(idMap); nodeID++ {
+				if err := testEndpoint(ctx, client, urlMap[nodeID], ep, "", ep.getVerifyResponse(), l, multiVersionTest); err != nil {
+					return errors.Wrapf(err, "failed testing endpoint %s", ep.url)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func testEndpoint(
+	ctx context.Context,
+	client *roachtestutil.RoachtestHTTPClient,
+	baseURL string,
+	ep endpoint,
+	nodeID string,
+	verifyResponse func(*http.Response) error,
+	l *logger.Logger,
+	multiVersionTest bool,
+) error {
+	fullURL := baseURL + ep.url
+	var err error
+	fullURL, err = fillPlaceholders(fullURL, nodeID)
+	if err != nil {
+		return err
+	}
+
+	l.Printf("testing endpoint: %s", fullURL)
+	var resp *http.Response
+	switch ep.method {
+	case GET:
+		resp, err = client.Get(ctx, fullURL)
+	default:
+		return errors.Newf("unsupported HTTP method: %s", ep.method)
+	}
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to %s %s", ep.method, fullURL)
+	}
+	defer resp.Body.Close()
+
+	// Note(alyshan): In multiversion tests, some endpoints may be unavailable
+	// as we may be acting as a secondary tenant (401, 501), or the database is on a version
+	// that does not support the endpoint (409).
+	if multiVersionTest {
+		// Since nodes get restarted in multiversion tests there is a race where certain nodes have
+		// not registered certain routes yet. Retry a few times to handle transient 404s.
+		if resp.StatusCode == http.StatusNotFound {
+			r := retry.StartWithCtx(ctx, retry.Options{
+				MaxRetries: 10,
+			})
+			for r.Next() {
+				resp, err = client.Get(ctx, fullURL)
+				if err != nil {
+					return errors.Wrapf(err, "failed to %s %s", ep.method, fullURL)
+				}
+				if resp.StatusCode != http.StatusNotFound {
+					break
+				}
+			}
+		}
+
+		// TODO(alyshan): Consider explicitly noting which endpoints fall into this category.
+		if resp.StatusCode == http.StatusUnauthorized || resp.StatusCode == http.StatusNotImplemented ||
+			resp.StatusCode == http.StatusConflict {
+			l.Printf("%v returned %v in multiversion testing mode", fullURL, resp.StatusCode)
+			return nil
+		}
+	}
+
+	err = verifyResponse(resp)
+	if err != nil {
+		return errors.Wrap(err, "failed to verify response")
+	}
+
+	return nil
+}
+
+func initializeSchemaAndIDs(ctx context.Context, c cluster.Cluster, l *logger.Logger) error {
```

Comment:
Could we get a docstring for this function? Despite it not being public, I feel like it's pretty important to the operation of this test.